// hydrogen-cli/lib/services/file/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const getPages_1 = tslib_1.__importDefault(require("./getPages"));
exports.getPages = getPages_1.default;
const getLayouts_1 = tslib_1.__importDefault(require("./getLayouts"));
exports.getLayouts = getLayouts_1.default;
const saveHTMLToFiles_1 = tslib_1.__importDefault(require("./saveHTMLToFiles"));
exports.saveHTMLToFiles = saveHTMLToFiles_1.default;
const getConfig_1 = tslib_1.__importDefault(require("./getConfig"));
exports.getConfig = getConfig_1.default;
const copyStaticFolder_1 = tslib_1.__importDefault(require("./copyStaticFolder"));
exports.copyStaticFolder = copyStaticFolder_1.default;
const copyExtraStaticFiles_1 = tslib_1.__importDefault(require("./copyExtraStaticFiles"));
exports.copyExtraStaticFiles = copyExtraStaticFiles_1.default;
const getServiceWorker_1 = tslib_1.__importDefault(require("./getServiceWorker"));
exports.getServiceWorker = getServiceWorker_1.default;
const deleteDistFolder_1 = tslib_1.__importDefault(require("./deleteDistFolder"));
exports.deleteDistFolder = deleteDistFolder_1.default;
const getHooks_1 = tslib_1.__importDefault(require("./getHooks"));
exports.getHooks = getHooks_1.default;
