// hydrogen-cli/lib/services/template/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
const head_1 = require("../head");
const file_1 = require("../file");
const LAYOUT_DEFAULT_TEMPLATE = 'default';
exports.mergeLayoutsWithPages = (pages, layouts) => pages
    .map((_a) => {
    var { layout = LAYOUT_DEFAULT_TEMPLATE } = _a, otherValues = tslib_1.__rest(_a, ["layout"]);
    return (Object.assign({ layout: layouts.filter(({ name }) => name === layout)[0].default }, otherValues));
});
exports.generateHTML = (pages, { dev, config }) => Promise.all(pages.map(async (page) => {
    var _a, _b, _c;
    const hooks = await file_1.getHooks();
    await ((_c = (_a = hooks) === null || _a === void 0 ? void 0 : (_b = _a).beforeEachPageGenerated) === null || _c === void 0 ? void 0 : _c.call(_b, { page, dev, config }));
    const pageData = page.data ? await page.data({ dev, config, route: page.route }) : {};
    const pageHead = head_1.transformHeadToHTML({
        head: page.head,
        data: pageData,
        // @ts-ignore
        route: page.route,
        config,
        dev,
    });
    const pageTemplate = await page.page({
        data: pageData,
        config,
        dev,
        route: page.route,
    });
    const layoutTemplate = await page.layout({
        title: page.title,
        content: pageTemplate,
        head: await pageHead,
        config,
        dev,
    });
    return {
        html: layoutTemplate,
        name: page.name.replace('js', 'html'),
        path: page.path,
    };
}));
